app/src/main/java/com/vasilevkin/catsanddogs/views/MainActivity.kt

```diff
@@ -0,0 +1,142 @@
+package com.vasilevkin.catsanddogs.views
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.vasilevkin.catsanddogs.R
+import com.vasilevkin.catsanddogs.features.animalList.CatAdapter
+import com.vasilevkin.catsanddogs.models.networkModels.CatImageRemoteModel
+import com.vasilevkin.catsanddogs.models.networkModels.CatRemoteModel
+import com.vasilevkin.catsanddogs.utils.getDataServiceCommon
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.internal.schedulers.IoScheduler
+import kotlinx.android.synthetic.main.activity_main.*
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.core.app.ComponentActivity
+import androidx.core.app.ComponentActivity.ExtraData
+import androidx.core.content.ContextCompat.getSystemService
+import android.icu.lang.UCharacter.GraphemeClusterBreak.T
+
+
+
+
+class MainActivity : AppCompatActivity() {
+
+    private var cats: List<CatImageRemoteModel> = emptyList()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+
+        val manager = GridLayoutManager(this, 2)
+        manager.spanSizeLookup = object : GridLayoutManager.SpanSizeLookup() {
+            override fun getSpanSize(position: Int): Int {
+                return if (position % 3 == 0) {
+                    2
+                } else {
+                    1
+                }
+            }
+        }
+        catList.layoutManager = manager
+//        catList.layoutManager = LinearLayoutManager(this)
+
+        val catsApi = getDataServiceCommon()
+
+        val response = catsApi.getBreedWithImages()
+//        val response = catsApi.getAllBreeds()
+
+        response
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeOn(IoScheduler())
+            .subscribe({ cats ->
+                this.cats = cats
+                catList.adapter = CatAdapter(cats, this)
```

Comment:
At the moment you create new adapter when every data received, you should update items in your adapter

Reply:
Yes, you're right, I see.
Could you please explain how to fix this?

Reply:
In your case, you should init you adapter early as adapter and crate method in the adapter that changes your data like this.cats = cats, also you can invoke adapter.notifyItemsSetChanged()

Reply:
https://www.semicolonworld.com/question/45552/how-to-update-recyclerview-adapter-data